wpimath/src/main/java/edu/wpi/first/math/kinematics/SwerveDriveKinematics.java

```diff
@@ -233,7 +233,11 @@ public Twist2d toTwist2d(SwerveModulePosition... wheelDeltas) {
    */
   public static void desaturateWheelSpeeds(
       SwerveModuleState[] moduleStates, double attainableMaxSpeedMetersPerSecond) {
-    double realMaxSpeed = Collections.max(Arrays.asList(moduleStates)).speedMetersPerSecond;
+    var absoluteModuleSpeeds = new Double[moduleStates.length];
+    for (int i = 0; i < absoluteModuleSpeeds.length; i++) {
+      absoluteModuleSpeeds[i] = Math.abs(moduleStates[i].speedMetersPerSecond);
+    }
+    double realMaxSpeed = Collections.max(Arrays.asList(absoluteModuleSpeeds));
```

Comment:
if you are already doing a for loop, don't call `Collections::max` and just do `Math::max` over `absoluteModuleSpeeds[i]` and define realMaxSpeed before the loop with a value of 0.

Reply:
Even better, `absoluteModuleStates` isn't even needed, as `realMaxSpeed` will store the max speed as the module states are iterated over, meaning that the array isn't even needed, and we can just use `moduleStates` instead. It also means we can use an enhanced for loop.